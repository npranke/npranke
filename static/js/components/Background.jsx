import React from 'react'

import Landscape from '@img/background-landscape.jpg'
import Portrait from '@img/background-portrait.jpg'

function Background() {
    return (
        <picture>
            <source srcSet={ Portrait } media="(orientation: portrait)" />
            <img
                src={ Landscape }
                className="background"
                alt="Conifer in Colorado landscape"
            />
        </picture>
    )
}

export default Background
