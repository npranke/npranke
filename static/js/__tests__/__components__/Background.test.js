import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import Background from '@components/Background'

Enzyme.configure({ adapter: new Adapter() })

describe('Background', () => {
    test('has alt attribute', () => {
        const background = shallow(<Background />)

        expect(
            background.find('.background').props().alt,
        ).toBeTruthy()
    })
})

describe('Background snapshot', () => {
    test('matches snapshot', () => {
        const background = shallow(<Background />)

        expect(background).toMatchSnapshot()
    })
})
