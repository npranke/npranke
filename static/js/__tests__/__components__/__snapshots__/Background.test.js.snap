// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background snapshot matches snapshot 1`] = `
<DocumentFragment>
  <picture>
    <source
      media="(orientation: portrait)"
      srcset="image-mock"
    />
    <img
      alt="Conifer in Colorado landscape"
      class="background"
      src="image-mock"
    />
  </picture>
</DocumentFragment>
`;
